src/gluonts/evaluation/new_metrics.py

```diff
@@ -0,0 +1,230 @@
+
+    def loss(self, x):
+        return 2 * np.abs(x.loss * ((x.target <= x.forecast) - self.q))
+
+
+@dataclass
+class WeightedQuantileLoss(DerivedMetric):
+    quantile: float
+
+    @property
+    def name(self):
+        return f"wQuantileLoss[{self.quantile}]"
+
+    @property
+    def dependencies(self):
+        return [QuantileLoss(self.quantile), AbsTargetSum()]
+
+    def get(self, metrics):
+        ql = QuantileLoss(self.quantile)
+
+        return metrics[ql.name] / metrics.abs_target_sum
+
+
+@dataclass
+class MeanWeightedQuantileLoss(DerivedMetric):
+    quantiles: List[float] = (0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9)
+    name: str = "mean_wQuantileLoss"
+
+    @property
+    def dependencies(self):
+        return list(map(WeightedQuantileLoss, self.quantiles))
+
+    def get(self, metrics):
+        return np.mean(
+            [metrics[quantile.name] for quantile in self.dependencies],
+            axis=0,
+        )
+
+
+def merge(a, b):
+    """Only merge items from b if they are not in a.
+
+    This is needed so that the order of `a` is not altered.
+    """
+    for key, val in dict.items(b):
+        if key not in a:
+            a[key] = val
+
+    return a
+
+
+def resolve(entity):
+    entities = {}
+
+    for dep in map(resolve, entity.dependencies):
+        merge(entities, dep)
+
+    entities[entity.name] = entity
+
+    return entities
+
+
+def get_order(entities):
+    result = {}
+
+    for entity in entities:
+        merge(result, resolve(entity))
+
+    return result
+
+
+def calc_aggrs(aggr_fns, data, axis=None):
+    result = AttrDict({})
+
+    for aggr in aggr_fns:
+        result[aggr.name] = aggr.get_aggr(data, result, axis=axis)
+
+    return result
+
+
+def calc_losses(loss_fns, data):
+    result = AttrDict({})
+
+    for loss in loss_fns:
+        loss.apply(data)
+
+    return data
+
+
+def select_metrics(names, data):
+    return {name: value for name, value in dict.items(data) if name in names}
+
+
+def get_evaluator(metrics):
+    ops = get_order(metrics)
+
+    losses = [op for op in ops.values() if isinstance(op, (Metric, Stat))]
+
+    aggrs = [
+        op for op in ops.values() if isinstance(op, (Metric, DerivedMetric))
+    ]
+
+    return (
+        partial(calc_losses, losses),
+        partial(calc_aggrs, aggrs),
+        partial(select_metrics, [metric.name for metric in metrics]),
+    )
+
+
+class AttrDict(dict):
+    def __getattribute__(self, key):
+        return self[key]
+
+    def __setattr__(self, key, value):
+        self[key] = value
+
+
+data = {
+    "target": np.array(
+        [
+            [1, 2, 3, 4, 5],
+            [1, 2, 3, 4, 5],
+        ]
+    ),
+    "forecast": np.array(
+        [
+            [2, 2, 2, 5, 5],
+            [1, 2, 3, 4, 5],
+        ]
+    ),
+}
+
+
+get_losses, aggregate, select = get_evaluator([MeanWeightedQuantileLoss()])
+
+value_metrics = get_losses(AttrDict(data))
+item_metrics = aggregate(value_metrics, axis=1)
```

Comment:
Since the aggregation of data seems to be `axis` dependant, can we include the possibility of computing metrics per _time stamp_? For instance, if we have data in daily frequency, can we compute the metrics per day?

Reply:
Yes, you need to align the data and select the axis accordingly.